src/cares_wrap.cc

```diff
@@ -1935,6 +1935,30 @@ void IsIPv6(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void canonicalIP(const FunctionCallbackInfo<Value>& args) {
+  node::Utf8Value ip(args.GetIsolate(), args[0]);
+  char address_buffer[sizeof(struct in6_addr)];
+  v8::Isolate* isolate = args.GetIsolate();
+
+  if (uv_inet_pton(AF_INET, *ip, &address_buffer) == 0) {
+    char canonical_ip[INET_ADDRSTRLEN];
+
+    int err = uv_inet_ntop(AF_INET, address_buffer, canonical_ip,
+      sizeof(canonical_ip));
+    CHECK_EQ(err, 0);
+
+    args.GetReturnValue().Set(String::NewFromUtf8(isolate, canonical_ip));
+  } else if (uv_inet_pton(AF_INET6, *ip, &address_buffer) == 0) {
+    char canonical_ip[INET6_ADDRSTRLEN];
```

Comment:
I'd just move this up a scope, it's big enough to hold IPv4 and IPv6 address strings.  Shaves off a few lines of code.